### ISSUE TITLE
`jj evolog` should show the reason each commit was rewritten

### ISSUE THREAD

**Issue by ilyagr** [jj-vcs/jj#963]
Currently, without `--patch`, `jj obslog` shows very little useful information -- only the commit id changes between the entries.

Similarly to the operation log, it should say e.g. "rewritten by `jj rebase -s blah -d bleh`". In fact, it could simply show the operation log entry that rewrote it.

**Comment by martinvonz** [comment]
Yes, I would like to make `jj obslog` walk the operation log instead of walking the predecessors. We would then need to filter the operation log to show only operations that touched the change (i.e. the thing identified by a change ID). That's currently pretty slow, IIRC from experiments I did, so we'd probably need to index it somehow.

A functional difference if we rewrite `jj obslog` that way is that we'd also include e.g. `jj undo` operations in the output, for better or worse. To clarify, if you do `jj new; jj describe -m foo; jj undo; jj obslog` today, you'd see a single node in the log; if we use the operation log instead, we'd include three nodes (the first and the last would be the same commit id).

**Comment by ilyagr** [comment]
That's a very good point about `undo`. IMO, it's not at all desirable to have `obslog` show undo operations, but I don't know if there's a version of this feature where that could be avoided.
